MTFO/Managers/CustomPuzzleManager.cs

```diff
@@ -0,0 +1,150 @@
+﻿using ChainedPuzzles;
+using UnityEngine;
+using static MTFO.Custom.CustomBioScan;
+using MTFO.Custom;
+using MTFO.Utilities;
+using System;
+using System.Collections.Generic;
+
+namespace MTFO.Managers
+{
+    /// <summary>
+    /// Handles all custom puzzle related stuff so it
+    /// isn't in a single patch like c'mon Dak you
+    /// had one job.
+    /// </summary>
+    public static class CustomPuzzleManager
+    {
+        private readonly static Dictionary<uint, CustomBioScan> s_scanMap = new();
+        private readonly static Dictionary<uint, CustomClusterScan> s_clusterMap = new();
+
+        public static bool TryGetScanByID(uint id, out CustomBioScan scan)
+            => s_scanMap.TryGetValue(id, out scan);
+
+        public static bool TryGetClusterByID(uint id, out CustomClusterScan cluster)
+            => s_clusterMap.TryGetValue(id, out cluster);
+
+        #region Initialization
+
+        private static bool s_initialized = false;
+
+        internal static void Initialize(ChainedPuzzleManager manager)
+        {
+            if (s_initialized) return;
+            s_initialized = true;
+
+            if (ConfigManager.CustomContent.ScanHolder != null)
+            {
+                InitScans(manager);
+                InitClusters(manager);
+            }
+        }
+
+        private static void InitScans(ChainedPuzzleManager manager)
+        {
+            if (manager == null)
+            {
+                Log.Error("Attempted to initialize custom bioscans with a null ChainedPuzzleManager. This should not happen!");
+                throw new NullReferenceException("ChainedPuzzleManager was null");
+            }
+
+            foreach (var scan in ConfigManager.CustomContent.ScanHolder.Scans)
+            {
+                Log.Debug($"Adding scan with ID of [{scan.PersistentID}]");
+                if (!manager.m_puzzleComponentPrefabs.ContainsKey(scan.BaseScan))
+                {
+                    Log.Error($"Custom scan with persistent ID {scan.PersistentID} references a non-existing base scan of persistent ID {scan.BaseScan}");
+                    continue;
+                }
+
+                //Base instance
+                var scanBase = manager.m_puzzleComponentPrefabs[scan.BaseScan];
+                var scanPrefab = GameObject.Instantiate(scanBase);
+                // why
```

Comment:
There used to be a comment here explaining why.

Something to do with the scan having to be moved far away because it rendered for a couple sevonds(?)

Probably should explain why instead of asking why.. its your rewrite